bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -109,53 +147,72 @@ public RuleRegistryImpl() {
      *
      * @param componentContext this component context.
      */
-
+    @Activate
     protected void activate(BundleContext bundleContext, Map<String, Object> properties) throws Exception {
-        ruleEngine.setCompositeModuleHandlerFactory(
-                new CompositeModuleHandlerFactory(bundleContext, moduleTypeRegistry, ruleEngine));
-        ruleEngine.setStatusInfoCallback(this);
         modified(properties);
         super.activate(bundleContext);
     }
 
+    /**
+     * This method is responsible for updating the value of delay between rule's re-initialization tries.
+     *
+     * @param config a {@link Map} containing the new value of delay.
+     */
+    @Modified
     protected void modified(Map<String, Object> config) {
-        ruleEngine.scheduleRulesConfigurationUpdated(config);
+        Object value = config == null ? null : config.get(CONFIG_PROPERTY_REINITIALIZATION_DELAY);
+        this.scheduleReinitializationDelay = (value != null && value instanceof Number) ? (((Number) value).longValue())
+                : DEFAULT_REINITIALIZATION_DELAY;
+        if (value != null && !(value instanceof Number)) {
+            logger.error("Invalid configuration value: {}. It MUST be Number.", value);
```

Comment:
I would prefer a warning and going with the default value then - which means that the runtime can easily recover from the invalid config value.

Reply:
OK